formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/FormstackSarHandler.scala

```diff
@@ -0,0 +1,47 @@
+package com.gu.identity.formstackbatonrequests
+
+import java.util.UUID.randomUUID
+
+import com.gu.identity.formstackbatonrequests.BatonModels.{Completed, Failed, Pending, SarInitiateRequest, SarInitiateResponse, SarPerformRequest, SarRequest, SarResponse, SarStatusRequest, SarStatusResponse}
+import com.gu.identity.formstackbatonrequests.lambda.LambdaClient
+import com.gu.identity.formstackbatonrequests.s3.S3Client
+import com.typesafe.scalalogging.LazyLogging
+
+case class FormstackSarHandler(s3Client: S3Client, lambdaClient: LambdaClient, sarHandlerConfig: SarLambdaConfig)
+  extends LazyLogging
+    with FormstackHandler[SarRequest, SarResponse] {
+
+  private def initiate(request: SarInitiateRequest): Either[Throwable, SarInitiateResponse] = {
+    val initiationReference = randomUUID.toString
+
+    val performSarRequest = SarPerformRequest(
+      initiationReference,
+      request.subjectEmail
+    )
+
+    logger.info(s"invoking FormstackPerformSarLambda with initiation reference: $initiationReference")
+    lambdaClient.invokeLambda(performSarRequest, sarHandlerConfig)
+      .map(_ => SarInitiateResponse(initiationReference))
+  }
```

Comment:
I know this is Scala but unpacking a map and `_` magic feels like overkill. I'm not best placed to have an opinion

But whats different to

```suggestion
    lambdaClient.invokeLambda(performSarRequest, sarHandlerConfig)
    SarInitiateResponse(initiationReference)
  }
```

Reply:
Well this function needs to return  `Either[Throwable, SarInitiateResponse]` so we cannot just return a `SarInitiateResponse`. 
InvokeLambda returns an `Either[Throwable, InvokeResult]` so mapping gets the correct return type.
This mapping is equivalent to just doing:
```scala
lambdaClient.invokeLambda(performSarRequest, sarHandlerConfig) match {
case Right(_) => SarInitiateResponse(initiationReference) 
case Left(error) => Left(error)
```
the `_` basically is there because it looks like in this case we don't really care what the successful response was, we are satisfied with it not returning an error

Reply:
👍 explained better than I could have